albumentations/augmentations/transforms.py

```diff
@@ -2664,3 +2667,151 @@ def get_params_dependent_on_targets(self, params: Dict[str, Any]) -> Dict[str, A
 
     def get_transform_init_args_names(self) -> Tuple[str, str, str, str, str, str, str]:
         return "mean", "std", "gauss_sigma", "intensity", "cutout_threshold", "mode", "color"
+
+
+class UnpropShuffle(DualTransform):
+    """
+    Unproportional shuffle grid's cells on image.
+    Args:
+        grid ((int, int)): size of grid for splitting image.
+    Targets:
+        image, mask
+    Image types:
+        uint8, float32
+    """
+
+    def __init__(self, grid=(3, 3), always_apply=False, p=0.5):
+        super(UnpropShuffle, self).__init__(always_apply, p)
+        self.grid = grid
+        self.shuffled_ids = []
+
+    def apply(self, img, tiles=None, **params):
+        if tiles is None:
+            tiles = []
+        self.ids = self.shuffled_ids.copy()
+        return F.unprop_swap_tiles_on_image(img, tiles, self.shuffled_ids)
+
+    def apply_to_mask(self, img, tiles=None, **params):
+        if tiles is None:
+            tiles = []
+        return F.unprop_swap_tiles_on_image(img, tiles, self.shuffled_ids)
```

Comment:
If it's the same logic as `apply`, remove it - by default for masks `apply` will be called.

Reply:
dbb4581811217408772c85a282446e4786b25d33